gapic/src/main/com/google/gapid/perfetto/views/TraceConfigDialog.java

```diff
@@ -492,8 +494,26 @@ public BasicInputArea(Composite parent, Settings settings, Theme theme,
               withIndents(new GridData(), GROUP_INDENT, 0));
           gpuCountersLabels = new Label[3];
           gpuCountersLabels[1] = createLabel(counterGroup, "Poll Rate:");
-          gpuCountersRate = createSpinner(counterGroup, sGpu.getCounterRate(), 1, 1000);
-          gpuCountersLabels[2] = createLabel(counterGroup, "ms");
+
+          long minSamplingPeriod = NANOSECONDS.toMicros(gpuCaps.getGpuCounterDescriptor().getMinSamplingPeriodNs());
+          long maxSamplingPeriod = NANOSECONDS.toMicros(gpuCaps.getGpuCounterDescriptor().getMaxSamplingPeriodNs());
+          maxSamplingPeriod = maxSamplingPeriod > 0 ? maxSamplingPeriod : minSamplingPeriod * 1000;
```

Comment:
what if the min sampling period is also 0? we should first set the min sampling period to 1ms if it's not specified.

Reply:
You are right. I plan to add validation and then notify vendors although I think they all at least advertise min sampling period. In that case do we still need to set the min sampling period to a default value?

Reply:
ping for feedback

Reply:
Maybe we can have a min value for now and add a todo for when the validation part is completed. When it has become a requirement that they should set min value, we can remove the hack then.

Reply:
I wouldn't call it a hack. We have a default for the max, it makes sense to have a default for the min. I would prefer a default we expect to never use over bad behavior we expect to never happen.